src/pages/PromptPage/PromptPage.jsx

```diff
@@ -1,99 +1,101 @@
+import { useEffect, useContext } from "react";
 import "./PromptPage.scss";
-import { useNavigate } from "react-router-dom";
+
 import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
 import Button from "../../components/Button/Button";
-import CoffeeMugWithHat_happy from "../../assets/images/coffeeMugWithHat_happy.svg";
-import { useEffect, useState } from "react";
-import QuizPage from "../../components/QuizPage/QuizPage";
-import Roadmap from "../../components/Roadmap/Roadmap";
-import GoalAchieved from "../../components/GoalAchieved/GoalAchieved";
-import LoadingPage from "../../components/LoadingPage/LoadingPage";
+import { PageContext } from "../../contexts/PageContext";
+import OnboardingPage1 from "../OnboardingPage/OnboardingPage1";
+import OnboardingPage2 from "../OnboardingPage/OnboardingPage2";
+import OnboardingPage3 from "../OnboardingPage/OnboardingPage3";
+import LoadingPage from "../LoadingPage/LoadingPage";
+import loadingInactiveIcon from "../../assets/images/loadingInactiveIcon.svg";
+import loadingActiveIcon from "../../assets/images/loadingActiveIcon.svg";
 
 const PromptPage = () => {
-  const navigate = useNavigate();
-  const [currentPage, setCurrentPage] = useState("prompt");
-
-  // ACCESS MATCH ALGO RESULTS through a state passed into quizPage
-  const [matchResults, setMatchResults] = useState(null);
-
+  const {
+    currentPage,
+    setCurrentPage,
+    progressBarIndex,
+    handleNext,
+    handleBack,
+  } = useContext(PageContext);
   useEffect(() => {
     setCurrentPage("prompt");
```

Comment:
You see here you could be setCurrentPageindex and then the content provider will let you know what the page name is instead. So you don't have the strings scattered in multiple pages.